app/controllers/candidate_interface/sign_in_controller.rb

```diff
@@ -56,6 +56,8 @@ def authenticate
           flash[:warning] = "You cannot have more than 3 course choices. You must delete a choice if you want to apply to #{course.name_and_code}."
           redirect_to candidate_interface_course_choices_review_path
         end
+      elsif !FindCandidateByToken.token_not_expired?(candidate)
+        redirect_to candidate_interface_expired_sign_in_path(id: candidate.id)
       else
```

Comment:
Will this `else` still run now? 🤔 We are covering both true/false for the `FindCandidateByToken.token_not_expired?` call, so it seems like this is dead code. Doesn't feel like it should be...

Reply:
Ah good spot! I think in my head I thought it was would catch when a dodgy token or when a token isn't provided but that is totally not the case like you've spotted.

I just tested and it blows up at the that `elsif` as you get a no method error for `id` on `candidate` which makes sense.

So it should be:

```ruby
def authenticate
  candidate = FindCandidateByToken.call(raw_token: params[:token])

  if candidate.nil?
    redirect_to action: :new
  elsif FindCandidateByToken.token_not_expired?(candidate)
    flash[:success] = t('apply_from_find.account_created_message') if candidate.last_signed_in_at.nil?

    sign_in(candidate, scope: :candidate)

    # ...
  else
    redirect_to candidate_interface_expired_sign_in_path(id: candidate.id)
  end
end
```

Reply:
That makes sense! 👍